tests/rptest/tests/read_distribution_test.py

```diff
@@ -0,0 +1,85 @@
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+from rptest.clients.kafka_cat import KafkaCat
+from rptest.services.redpanda import MetricsEndpoint
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class ReadDistributionTest(RedpandaTest):
+    def __init__(self, test_context):
+        super().__init__(test_context=test_context, num_brokers=1)
+
+        self.topic_name = 'tapioca'
+        self.topics = [TopicSpec(name=self.topic_name, replication_factor=1)]
+
+        self.record_size = 1024
+        self.record_count = 10
+        self.records_per_batch = 2
+        self.timespan_ms = 24 * 60 * 60 * 1000  #One day in ms.
+        self.timestamp_step_ms = self.timespan_ms // (self.record_count - 1)
+        self.curr_ts = round(time.time() * 1000)
+        self.base_ts = self.curr_ts - self.timespan_ms  #Start producing from record_count days ago.
+
+    @cluster(num_nodes=2)
+    def test_read_distribution_metric(self):
+        '''
+        Validate that Kafka reads are being correctly set in the read distribution histogram.
+        '''
+        KgoVerifierProducer.oneshot(
+            self.test_context,
+            self.redpanda,
+            self.topic_name,
+            self.record_size,
+            self.record_count,
+            batch_max_bytes=self.record_size * self.records_per_batch - 1,
+            fake_timestamp_ms=self.base_ts,
+            fake_timestamp_step_ms=self.timestamp_step_ms)
+        kcat = KafkaCat(self.redpanda)
+        for i in range(0, self.record_count):
+            timestamp = self.base_ts + i * self.timestamp_step_ms
+            kcat.consume_one(self.topic_name,
+                             0,
+                             offset=None,
+                             first_timestamp=timestamp)
+
+        metrics = self.redpanda.metrics(self.redpanda.nodes[0],
+                                        MetricsEndpoint.PUBLIC_METRICS)
+        read_dist_metric = [
+            m for m in metrics if m.name == 'redpanda_kafka_read_distribution'
+        ][0]
+        cumulative_topic_read_dist = [
+            (s.value, s.labels['le']) for s in read_dist_metric.samples if
+            s.labels['redpanda_topic'] == self.topic_name and 'le' in s.labels
+        ]
+        prev = 0
+        topic_read_dist = {}
+        for v, le in cumulative_topic_read_dist:
+            try:
+                topic_read_dist[int(float(le))] = int(v - prev)
+                prev = v
+            except:
+                continue
+
+        expected_read_dist = {bucket: 0 for bucket in topic_read_dist.keys()}
+        bucket_list = list(topic_read_dist.keys())
+        for i in range(0, self.record_count):
+            t_minutes = round(
+                (self.curr_ts -
+                 (self.base_ts + i * self.timestamp_step_ms)) / (1000 * 60))
+            bucket = bucket_list[bisect.bisect_left(bucket_list, t_minutes)]
```

Comment:
nit: mind adding a comment explaining what we're doing here? On first read this is hard to follow

Reply:
Also, it kind of seems like it should be a fixed distribution since base_ts is curr_ts - timespan_ms. Given the bucketing will likely not change much, do you think it'd make sense to build the map manually with fixed values? IMO it'd be much easier to verify that the distribution is what we expect

Reply:
Hmm. This was actually my attempt to stay away from hardcoding any map values, just in case any of the test parameters or histogram bucketing values are changed later.

I think I prefer this way of checking the distribution, but if you feel strongly that we should build the map manually, I can make the change.

Added a comment for now explaining the `bisect_left` portion.

Reply:
In principal I agree with you, it makes the code adaptable to further bucketing/test changes. OTOH it introduces room for bugs in this test code. I'm fine with it being here, but as a reader I think it would help to add some hardcoded values (i.e maybe let's do both)

Reply:
Added some hardcoded values to the `read_distribution_test`.